src/com/serotonin/mango/rt/dataImage/PointValueCache.java

```diff
@@ -144,16 +174,25 @@ private void refreshCache(int size) {
     }
 
     public void reset() {
-        List<PointValueTime> c = cache;
 
         int size = defaultSize;
-        if (c.size() < size)
-            size = c.size();
 
-        List<PointValueTime> nc = new ArrayList<PointValueTime>(size);
-        nc.addAll(c.subList(0, size));
+        if (cache.size() < size)
+            size = cache.size();
 
         maxSize = size;
-        cache = c;
+    }
+
+    private void readOnlyOneRowWithMaxTSAndPutIntoCache(int dataPointId){
+
+        PointValueTime pointValueTime = getPointValueCacheCooperateWithPointValueDao().getLatestPointValueFromDao( dataPointId );
```

Comment:
Do you call the dao method fromDao?

Reply:
Yes, that's the separated class which has been created only to cooperation with dao.Dao methods  which were in PointValueCache shouldn't appeared here.

Reply:
I meant the name here. Everything in the DAO class means that it comes from dao, so you shouldn't need to use the word dao in method names. Basically, the cache should be an invisible layer, and I can put it in DAO as the layer closest to the database.